tests/unit/azure/test_locations.py

```diff
@@ -120,6 +114,83 @@ def test_run_service_principal(ws):
     )
 
 
+def test_run_unsupported_location(ws, caplog):
+    """test unsupported location will not be migrated"""
+
+    # mock crawled HMS external locations with two unsupported locations adl and wasbs
+    row_factory = type("Row", (Row,), {"__columns__": ["location", "table_count"]})
```

Comment:
Use MockBackend.rows() - see other tests where I replaced it recently: https://github.com/databrickslabs/lsql/commit/f03a5ab03338d38cc6bd29d6cd28d6c5c367a11d

Reply:
fixed